### DISCUSSION TITLE
Provide UI feedback if attempting to delete asset used in a collection

### DISCUSSION THREAD

**Discussion by nathanhere** [directus/directus#6271] [Feature Requests]
I attempted to unsuccessfully delete an asset, with the UI confirmation modal remaining on the screen after confirming I wanted to continue with deletion.  Looking at the logs, I saw the readout below:

```
error: delete from "directus_files" where "id" in ($1, $2, $3, $4) - update or delete on table "directus_files" violates foreign key constraint "review_image_foreign" on table "review"
```

I think it could be helpful to provide a modal with feedback to users if they are attempting to delete an asset currently being used in a collection, or some sort of visual indicator on the asset icon itself that shows if it is currently used in a collection.

**Comment by rijkvanzanten** [comment]
Step one would be to write an error abstraction (https://github.com/directus/directus/tree/main/api/src/exceptions/database) so we can show a proper error (inline) on save

**Comment by domdus** [comment]
I'm surprised, thougth it would actually just delete the files - good to know it prevents deletion if the file is linked to a collection item. (Is that new?)

Great idea anyway, with a visual indicator on files you  would be able to know - what files are save to be deleted (be able to filter-select them for deletion).

**Reply by rijkvanzanten** [reply]
> I'm surprised, thougth it would actually just delete the files - good to know it prevents deletion if the file is linked to a collection item. (Is that new?)

Directus supports foreign key constraints in the database, that was added in RC 66 I believe (between 60-70 at least) 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
